circuits/sybil/sybilResCredentialAtomicQuerySigOffChain.circom

```diff
@@ -0,0 +1,256 @@
+    verifyUniClaim.issuerAuthClaimNonRevMtpAuxHi <== issuerAuthClaimNonRevMtpAuxHi;
+    verifyUniClaim.issuerAuthClaimNonRevMtpAuxHv <== issuerAuthClaimNonRevMtpAuxHv;
+
+    for (var i=0; i<8; i++) { verifyUniClaim.issuerClaim[i] <== issuerClaim[i]; }
+
+    for (var i=0; i<IssuerLevels; i++) { verifyUniClaim.issuerClaimNonRevMtp[i] <== issuerClaimNonRevMtp[i]; }
+    verifyUniClaim.issuerClaimNonRevMtpNoAux <== issuerClaimNonRevMtpNoAux;
+    verifyUniClaim.issuerClaimNonRevMtpAuxHi <== issuerClaimNonRevMtpAuxHi;
+    verifyUniClaim.issuerClaimNonRevMtpAuxHv <== issuerClaimNonRevMtpAuxHv;
+    verifyUniClaim.issuerClaimNonRevClaimsRoot <== issuerClaimNonRevClaimsRoot;
+    verifyUniClaim.issuerClaimNonRevRevRoot <== issuerClaimNonRevRevRoot;
+    verifyUniClaim.issuerClaimNonRevRootsRoot <== issuerClaimNonRevRootsRoot;
+    verifyUniClaim.issuerClaimNonRevState <== issuerClaimNonRevState;
+
+    verifyUniClaim.issuerClaimSignatureR8x <== issuerClaimSignatureR8x;
+    verifyUniClaim.issuerClaimSignatureR8y <== issuerClaimSignatureR8y;
+    verifyUniClaim.issuerClaimSignatureS <== issuerClaimSignatureS;
+
+    component uniClaimSchemaHash = GetUniquenessSchemaHash();
+    verifyUniClaim.issuerClaimSchema <== uniClaimSchemaHash.schemaHash;
+    verifyUniClaim.profileNonce <== profileNonce;
+    verifyUniClaim.userGenesisID <== userGenesisID;
+    verifyUniClaim.claimSubjectProfileNonce <== claimSubjectProfileNonce;
+
+  
+    verifyUniClaim.claimHash ==> uniClaimHash;
+
+    component verifyStateSecret = VerifyAndExtractValStateSecret(HolderLevel, GistLevels);
+    for (var i=0; i<8; i++) { verifyStateSecret.claim[i] <== holderClaim[i]; }
+    for (var i=0; i<HolderLevel; i++) { verifyStateSecret.claimIssuanceMtp[i] <== holderClaimMtp[i]; }
+    verifyStateSecret.claimIssuanceClaimsRoot <== holderClaimClaimsRoot;
+    verifyStateSecret.claimIssuanceRevRoot <== holderClaimRevRoot;
+    verifyStateSecret.claimIssuanceRootsRoot <== holderClaimRootsRoot;
+    verifyStateSecret.claimIssuanceIdenState <== holderClaimIdenState;
+
+    verifyStateSecret.genesisID <== userGenesisID; 
+
+    for (var i=0; i<GistLevels; i++) { verifyStateSecret.gistMtp[i] <== gistMtp[i]; }
+    verifyStateSecret.gistRoot <== gistRoot;
+    verifyStateSecret.gistMtpAuxHi <== gistMtpAuxHi;
+    verifyStateSecret.gistMtpAuxHv <== gistMtpAuxHv;
+    verifyStateSecret.gistMtpNoAux <== gistMtpNoAux;
+
+    verifyStateSecret.secretClaimHash ==> secretClaimHash;
+    
+    // 3. Compute profile.
+    component selectProfile = SelectProfile();
+    selectProfile.in <== userGenesisID;
+    selectProfile.nonce <== profileNonce;
+    userID <== selectProfile.out;
+
+    // 4. Compute sybil
+    component computeSybilID = ComputeSybilID();
+    computeSybilID.crs <== crs;
+    computeSybilID.stateSecret <== secretClaimHash;
+    computeSybilID.claimHash <== uniClaimHash;
+    sybilID <== computeSybilID.out;
+}
+
+
+template VerifyAndHashUniClaim(IssuerLevels){
```

Comment:
Again, the essential of this code is repeated in the `lib/query/credentialAtomicQuerySigV2.circom`. Why do we have the same in 2 places?

Reply:
Good observation, I overlooked it.

At first glance they seem similar but I can see a few differences:

1. The circuit is accepting two signatures (challenge and claim), in my circuit, I'm working with just one signature like the [SigOffChain circuit](https://github.com/iden3/circuits/blob/feature/circuits_v0.2/circuits/offchain/credentialAtomicQuerySigOffChain.circom).
2. I don't need to use the query mechanizem.
3. I can see it also checks the GIST which I'm not. If that's not a must (its not?), let's keep it out to reduce constraints. (I see that I can change the template to support enabled flag for gist check).